File: solutions/lua/triangle/1/triangle.lua
local triangle = {}

function triangle.kind(a, b, c)
	if a <= 0 or b <= 0 or c <= 0 then
		error("Input Error")
	elseif a + b < c or b + c < a or a + c < b then
		error("Input Error")
	elseif a == b and b == c and c == a then
		return "equilateral"
	elseif a == b or b == c or c == a then
		return "isosceles"
	elseif a ~= b and b ~= c and c ~= a then
		return "scalene"
	end
end

return triangle

File: solutions/cpp/triangle/1/triangle.cpp
#include <stdexcept>
#include "triangle.h"

namespace triangle {

    flavor kind(float a, float b, float c) {
        if (0 == a || 0 == b || 0 == c) {
            throw std::domain_error("null sided triangle");
        }
        if (a > b + c || b > a + c || c > a + b) {
            throw std::domain_error("triangle inequality invalidadated");
        }
        if (a == b && b == c) {
            return flavor::equilateral;
        }
        if (a == b || b == c || a == c) {
            return flavor::isosceles;
        }
        return flavor::scalene;
    }

}  // namespace triangle
